docs/design/uninstall_osm.md

```diff
@@ -0,0 +1,66 @@
+# Uninstall OSM Control Plane from Cluster
+
+- **Revision:** 1
+- **Status:** Implemented
+
+## Table of Contents
+<!-- toc -->
+- [Summary](#summary)
+  - [Use Case](#use-case)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+  - [Experience](#experience)
+  - [Solution](#solution)
+    - [Implementation Details](#implementation-details)
+  - [Testing](#testing)
+- [Historical Context](#historical-context)
+- [Looking Forward](#looking-forward)
+<!-- /toc -->
+
+## Summary
+A user should be able to delete all Kubernetes resources related to an OSM control plane installation using a CLI command.
+
+### Use Case
+A user can use `osm install` to install an OSM control plane in a Kubernetes cluster to test out OSM. Once tested, they may want to uninstall the control plane components.
+
+## Motivation
+Make it easier to cleanup OSM control planes in test clusters
+
+### Goals
+Delete all Kubernetes resources associated with OSM control plane.
+
+### Non goals
+- This version of the proposal will not handle removing any CRDs since these are cluster-wide resources that may be used by other OSM control planes, service mesh implementations or tools.
+- This version of the proposal will not handle removing Envoy proxies from application pods. The user must re-start application Pods once the `osm` control plane has been deleted. Traffic will be interrupted.
```

Comment:
> Traffic will be interrupted.

Is it true that using a [PodDisruptionBudget](https://kubernetes.io/docs/concepts/workloads/pods/disruptions/) for highly-available workloads would mitigate any interruptions? If so, is that worth mentioning here?

Reply:
A PodDisruptionBudget would be helpful if we wanted to ensure a certain number of Pods are available by blocking voluntary disruptions like using `kubectl drain` / draining a node for repair or upgrade. In this case, a PodDisruptionBudget would not be able help deal with traffic interruption unfortunately. Traffic is interrupted because the Envoy Proxy remains in the Pod but doesn't have a management server to talk to.